pkg/dependencies/dependents_test.go

```diff
@@ -0,0 +1,413 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package dependencies
+
+import (
+	"context"
+	"fmt"
+	clients "github.com/guacsec/guac/internal/testing/graphqlClients"
+	"testing"
+	"time"
+
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+
+	"github.com/Khan/genqlient/graphql"
+)
+
+type testPkgInputSpec struct {
+	id       string
+	typeName string
+	name     string
+	version  string
+}
+
+type createIsDependency struct {
+	pkg    specInfo
+	depPkg specInfo
+}
+
+type specInfo struct {
+	name     string
+	version  string
+	typeName string
+}
+
+func Test_findAllDependencies(t *testing.T) {
+	gqlClient := clients.SetupTest(t)
```

Comment:
This should be called inside each subtest, otherwise each test may affect each other.

Reply:
Oh, thanks didn't think of that!